src/Plugin/Derivative/StrawberryLocalTasks.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\strawberryfield\Plugin\Derivative;
+
+use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\Core\Plugin\Discovery\ContainerDeriverInterface;
+use Drupal\form_mode_manager\FormModeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines dynamic local tasks.
+ */
+class StrawberryLocalTasks extends DeriverBase implements ContainerDeriverInterface {
```

Comment:
I see. So this is doing what content moderation does right?https://api.drupal.org/api/drupal/core%21modules%21content_moderation%21src%21Plugin%21Derivative%21DynamicLocalTasks.php/8.2.x

Since this Deriver only provides an Edit option, should the Task class name be more verbose about that? StrawberryLocalTasks.php  is quite generic. Or are you thinking of reusing this Deriver for any other need in the future?

Also form_mode_manager is not a dependency of the SBF module (see our composer.json) and i don't want to introduce it. 

Solution here is using core

Look at this piece of code 
```PHP
public function getFormModesByEntityTypeId($entity_type_id) {
    $form_modes = $this->entityDisplayRepository->getFormModes($entity_type_id);
    // this needs to be locally code of course and be replaced by something similar but simpler than \Drupal\form_mode_manager\FormModeManager::candidateToExclude
    $this->filterExcludedFormModes($form_modes, $entity_type_id, FALSE);
    return array_keys($form_modes);
  }
```

Basically, inject entityDisplayRepository instead of 'form_mode.manager' and use the core functionality. Your current call is just a wrapper and adds dependencies we don't need in this module.

Reply:
@DiegoPino Yes, a lot of modules follow this form and it's described in the docs:
https://www.drupal.org/docs/8/api/menu-api/providing-module-defined-local-tasks#dynamic-local-task-generation
I copied the approach of form mode manager module, basically.

But I will remove the dependency!